TaskfMRIAnalysis/scripts/makeSubjectTaskSummary.sh

```diff
@@ -0,0 +1,363 @@
+		else
+			cp -suv ${LevelTwoFEATDir}/Contrasts.txt ${SummaryDirectory}
+		fi
+		
+		copeCounter=1;
+		cat ${LevelTwoFEATDir}/Contrasts.txt | while read Contrast ; do
+			# Check if necessary files exist in cope${i}.feat directory (cope1 should be mean of two runs)
+			if [ -e "${LevelTwoFEATDir}/${Analysis}/cope${copeCounter}.feat/cope1.${Extension}" ]; then
+				echo "${Analysis}: cope${copeCounter} ($Contrast) exists" >> ${SummaryDirectory}/TaskfMRIAnalysis.log
+			fi
+			for File in {mask,cope1,varcope1,tdof_t1} ; do
+				cifti_in=${LevelTwoFEATDir}/${Analysis}/cope${copeCounter}.feat/${File}.${Extension}
+				if [ -e "$cifti_in" ]; then
+					outdir=${SummaryDirectory}/${Analysis}/cope${copeCounter}.feat
+					if [ ! -d "$outdir" ]; then
+						mkdir -pv $outdir;
+					fi
+					cifti_out=$(echo $cifti_in | sed -e "s|$LevelTwoFEATDir|$SummaryDirectory|")
+					# create symlink to summary directory
+					cp -suv $cifti_in $cifti_out
+				else
+					# file is missing, write cope number and contrast name to log
+					shortName=$( echo $cifti_in | sed -e "s|${ResultsFolder}/||" );
+					echo "${Analysis}: NOTE cope${copeCounter} ($Contrast) is missing file $shortName" >> ${SummaryDirectory}/TaskfMRIAnalysis.log
+				fi
+			done
+			copeCounter=$(($copeCounter+1))
+		done
+
+	else	# If Level2 not requested, determine location of Level1
+		echo "Using Level1"
+		# Cut first value from list (ensure single value is clean)
+		LevelOnefMRIName=$( echo $LevelOnefMRINames | cut -d'@' -f1 )
+		LevelOnefsfName=$( echo $LevelOnefsfNames | cut -d'@' -f1 )
+
+		# Determine which Level1 analysis was requested
+		LevelOneFEATDir="${ResultsFolder}/${LevelOnefMRIName}/${LevelOnefsfName}${TemporalFilterString}${SmoothingString}_level1${RegString}${ProcSTRING}${LowPassSTRING}${ParcellationString}.feat"
+
+		# Check if Level1 outputs are present, else throw error
+		if [ ! -e "${LevelOneFEATDir}/design.con" ]; then
+			# file is missing, write cope number and contrast name to log
+			echo "ERROR: Cannot find Contrasts list at ${LevelOneFEATDir}/design.con. Verify that Level1 analysis completed correctly." >> ${SummaryDirectory}/TaskfMRIAnalysis.log
+			log_Err_Abort "ERROR: Cannot find Contrasts list at ${LevelOneFEATDir}/design.con. Verify that Level1 analysis completed correctly."
+		fi
+
+		# Create mask and tdof files in tmp directory, to copy them into place as needed
+		tmpdir=${SummaryDirectory}/${Analysis}/tmp
+		mkdir -pv $tmpdir
+		# Symlink Level1 dof and create tdof_t1 image file and mask file
+		dof=`cat ${LevelOneFEATDir}/${Analysis}/dof`;
+		if [[ "$Analysis" = "GrayordinatesStats" || "$Analysis" = "ParcellatedStats" ]] ; then
+			${CARET7DIR}/wb_command -cifti-convert -to-nifti ${LevelOneFEATDir}/${Analysis}/res4d.${Extension} $tmpdir/res4d.nii.gz
+		else
+			cp -suv ${LevelOneFEATDir}/${Analysis}/res4d.nii.gz $tmpdir/res4d.nii.gz
+		fi
+		fslmaths $tmpdir/res4d.nii.gz -Tstd -bin $tmpdir/mask.nii.gz;
```

Comment:
Side note, if you want this `-Tstd -bin` to reliably exclude no-variance elements that aren't exactly zero, you should instead check whether the max and min aren't equal, as computing stdev inevitably contains floating point rounding error when run on anything that isn't zeros.